code/modules/power/cell.dm

```diff
@@ -103,9 +103,9 @@
 	return (FIRELOSS)
 
 /obj/item/stock_parts/cell/on_reagent_change(changetype)
-	rigged = !isnull(reagents.has_reagent(/datum/reagent/toxin/plasma, 5)) //has_reagent returns the reagent datum
-	..()
+	rigged = reagents.has_reagent(/datum/reagent/toxin/plasma, 5))

```

Comment:
rigged can be set by corrupt(), add handling for that so we don't override it as well.